scripts/micro_benchmarks.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python3
+
+'''
+Builds the Benchmarks
+'''
+
+from argparse import Action
+from argparse import ArgumentError
+from argparse import ArgumentParser
+from argparse import ArgumentTypeError
+from argparse import SUPPRESS
+from io import StringIO
+from logging import getLogger
+from os import path
+from subprocess import CalledProcessError
+from traceback import format_exc
+from typing import Tuple
+
+import csv
+import sys
+
+from performance.common import get_repo_root_path
+from performance.common import remove_directory
+from performance.common import validate_supported_runtime
+from performance.logger import setup_loggers
+
+import dotnet
+
+
+class TargetFrameworkAction(Action):
+    '''
+    Used by the ArgumentParser to represent the information needed to parse the
+    supported .NET Core target frameworks argument from the command line.
+    '''
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if values:
+            wrong_choices = []
+            supported_target_frameworks = TargetFrameworkAction\
+                .get_supported_target_frameworks()
+
+            for value in values:
+                if value not in supported_target_frameworks:
+                    wrong_choices.append(value)
+            if wrong_choices:
+                message = ', '.join(wrong_choices)
+                message = 'Invalid choice(s): {}'.format(message)
+                raise ArgumentError(self, message)
+            setattr(namespace, self.dest, list(set(values)))
+
+    @staticmethod
+    def get_supported_target_frameworks() -> list:
+        '''List of supported .NET Core target frameworks.'''
+        frameworks = list(
+            TargetFrameworkAction.__get_framework_channel_map().keys()
+        )
+        if sys.platform == 'win32':
+            frameworks.append('net461')
+        return frameworks
+
+    @staticmethod
+    def __get_framework_channel_map() -> dict:
+        # TODO: Can we do better? For example, read it from csproj?
+        return {
+            'netcoreapp3.0': 'master',
+            'netcoreapp2.2': 'release/2.2.2xx',
+            'netcoreapp2.1': 'release/2.1',
+            'netcoreapp2.0': 'release/2.0.0',
```

Comment:
with `release/2.0.0` it downloads `2.0.3-servicing-007156` and fails to build for netcoreapp2.0 with a really strange error:

```log
The specified framework 'Microsoft.NETCore.App', version '2.0.3' was not found.
```
the fix is to use `2.0` (I tested that it works)

```suggestion
            'netcoreapp2.0': '2.0',
```